net/net/inc/TSocket.h

```diff
@@ -80,7 +80,7 @@ friend class TSlave;       // to be able to call SendHostAuth()
 
    static Int_t  fgClientProtocol; // client "protocol" version
 
-   TSocket() : fAddress(), fBytesRecv(0), fBytesSent(0), fCompress(ROOT::RCompressionSetting::EAlgorithm::kUseGlobal),
+   TSocket() : fAddress(), fBytesRecv(0), fBytesSent(0), fCompress((Int_t)ROOT::RCompressionSetting::EAlgorithm::kUseGlobal),
```

Comment:
Here, we probably want to change the type of `fCompress` instead.

Reply:
Is this the case only for `TSocket` or also for the other classes that have a `Int_t fCompress` field?
Here is the list:
- `TFile`
- `TBufferSQL2`
- `TMessage`
- `TUDPSocket`
- `TBufferXML`
- `TBranch`
- `TProofServ`

Reply:
Actually I think it's not really doable here because this `int` encodes both the algorithm and the level, so it cannot be made into a `RCompressionSetting::EAlgorithm`.

Reply:
Right. I missed that (the class did not have `fLevel`) and the existing code is already a bit of abuse of the enum (I can't remember why we did not use here `ROOT::RCompressionSetting::EDefaults::kUseGlobal` (where the enum value are already the (algorithm,level) pair.

Reply:
Yeah, I think EDefaults would communicate the intention better. I'll change it.